mongodb/configure.sls

```diff
@@ -23,9 +23,9 @@ place_mongodb_config_file:
 
 {% if 'mongodb_primary' in grains['roles'] %}
 
-{% set replset_config = {'_id': salt['pillar.get']('mongodb:replset_name', 'rs0'), 'members': []} %}
+{% set replset_config = {'_id': salt.pillar.get('mongodb:replset_name', 'rs0'), 'members': []} %}
 {% if salt.pillar.get("mongodb:VAGRANT", false) %}
-{% do replset_config['members'].append({'_id': 0, 'host': '192.168.33.10:' + mongodb.port}) %}
+{% do replset_config['members'].append({'_id': 0, 'host': salt.pillar.get('mongodb:vagrant_host') + ':' + mongodb.port}) %}
```

Comment:
Since you set the `vagrant_host` variable in the map file then you can include that in the import at the top of the file and then use the variable reference here, or you can just pull it out of the `map.jinja` file and add the default value to this call to `pillar.get`.